src/normalizer.ts

```diff
@@ -333,7 +333,12 @@ class V1SchemaNormalizer {
       }
 
       s.type = 'object'
-      return this.annotateType(s.$ref)
+      let obj = this.annotateType(s.$ref)
+      if (obj && s.nullable) {
+        obj = {...obj}
+        obj.nullable = s.nullable
+      }
+      return obj
```

Comment:
Looks pretty clean. does this work? let's add a little note that this is a hack to fix the problem.

if this is ready let's also push a new version out. bump this to `rc17`: https://github.com/dylibso/xtp-bindgen/blob/main/package.json#L2
and run `npm install` (which should update the lock file). Then commit and push those up. I'll cut the release after merging.